From 6b6d8f4cebdbe001dc8b13de4bb78defe15bf87b Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Fri, 7 Feb 2020 14:43:22 +0530
Subject: [PATCH 470/767] octeontx2-pf: Don't configure RSS context when
 interface is down

commit 6248952f194687b88808ae5f2ed75d23a0faa9b6 from
git@git.assembla.com:cavium/WindRiver.linux.git

When interface is down all NIXLF contexts are disabled, so any
attempt to confgiure them will fail. Hence do not try to
configure RSS contexts when interface is down. Since the
user input is saved when interface comes up RSS is configured
as per user input.

Change-Id: I692761a125fd6e1fada78291e0124df5adc441d5
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/23059
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
index 60f02df8a33a..ea9b4b1cdbc5 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
@@ -929,7 +929,8 @@ static int otx2_set_rxfh(struct net_device *dev, const u32 *indir,
 		otx2_set_rss_key(pfvf);
 	}
 
-	otx2_set_rss_table(pfvf);
+	if (netif_running(dev))
+		otx2_set_rss_table(pfvf);
 	return 0;
 }
 
-- 
2.31.1

